generic/Tensor.c

```diff
@@ -1319,6 +1319,7 @@ void torch_Tensor_(init)(lua_State *L)
   luaT_setfuncs(L, torch_Tensor_(_), 0);
   lua_pop(L, 1);
   THVector_(vectorDispatchInit)();
+  THTensor_(tensorMathDispatchInit)();
```

Comment:
Hi, @apaszke , thanks for your kind help! Please see my recent changes.

I plan to merge `THTensorMathSIMD.c` into `SSE.c`, which you requested some days ago.

    If contiguous
      multi-thread fork-join
        THVector_op
    else
      TH_TENSOR_APPLY

I use THVector to apply the corresponding operation when data is contiguous. So in this paradigm, I merge all vector functions into `SSE.c` and `NEON.c` and maintained non-contiguous tensor operations. But notice that I also modify the original `add` function to `cadd` by doing this. 

If the design is ok, I will move on.